pic-sure-resources/pic-sure-aggregate-data-sharing-resource/src/main/java/edu/harvard/hms/dbmi/avillach/AggregateDataSharingResourceRS.java

```diff
@@ -163,31 +163,44 @@ public QueryStatus query(QueryRequest queryRequest) {
 
     }
 
-    @POST
-    @Path("/query/{resourceQueryId}/status")
-    @Override
-    public QueryStatus queryStatus(@PathParam("resourceQueryId") String queryId, QueryRequest statusRequest) {
-        logger.debug("Calling Aggregate Data Sharing Resource queryStatus() for query {}", queryId);
-        checkQuery(statusRequest);
-        HttpResponse response = postRequest(statusRequest, "/query/" + queryId + "/status");
-        return readObjectFromResponse(response, QueryStatus.class);
-    }
-
-    @POST
-    @Path("/query/{resourceQueryId}/result")
-    @Override
-    public Response queryResult(@PathParam("resourceQueryId") String queryId, QueryRequest resultRequest) {
-        logger.debug("Calling Aggregate Data Sharing Resource queryResult() for query {}", queryId);
-        checkQuery(resultRequest);
-        HttpResponse response = postRequest(resultRequest, "/query/" + queryId + "/result");
-        try {
-            return Response.ok(response.getEntity().getContent()).build();
-        } catch (IOException e) {
-            throw new ApplicationException(
-                    "Error encoding query for resource with id " + resultRequest.getResourceUUID()
-            );
-        }
-    }
+	@POST
+	@Path("/query/{resourceQueryId}/status")
+	@Override
+	public QueryStatus queryStatus(@PathParam("resourceQueryId") String queryId, QueryRequest statusRequest) {
+		// Validate the queryId
+		try {
+			UUID.fromString(queryId);
+		} catch (IllegalArgumentException e) {
+			throw new ProtocolException(ProtocolException.INCORRECTLY_FORMATTED_REQUEST);
+		}
```

Comment:
I can turn this into a util method since I am using the same logic a number of times.